core/chains/evm/client/errors.go

```diff
@@ -167,13 +167,16 @@ var klaytn = ClientErrors{
 // Nethermind
 // All errors: https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.TxPool/AcceptTxResult.cs
 // All filters: https://github.com/NethermindEth/nethermind/tree/9b68ec048c65f4b44fb863164c0dec3f7780d820/src/Nethermind/Nethermind.TxPool/Filters
-var nethermindFatal = regexp.MustCompile(`(: |^)(SenderIsContract|Invalid|Int256Overflow|FailedToResolveSender|GasLimitExceeded)$`)
+var nethermindFatal = regexp.MustCompile(`(: |^)(SenderIsContract|Invalid(, transaction Hash is null)?|Int256Overflow|FailedToResolveSender|GasLimitExceeded(, Gas limit: \d+, gas limit of rejected tx: \d+)?|NonceGap(, Future nonce. Expected nonce: \d+)?)$`)
```

Comment:
Instead of: SenderIsContract|Invalid(, transaction Hash is null)?
Does it make sense to just have: SenderIsContract|Invalid?

By trying to match the bracket part, are we getting anything extra?

Happy to have you hear from others too on their opinion.

Same comment on all cases below which have the bracketed() part added.

Reply:
We have hesitated to make the matches too open ended in the past, specifically prefixes that match any suffix. So we usually enumerate the possibilities and use the trailing `$`, like here. That internal one looks a little more complicated actually.

Reply:
In this case, if you look at the Nethermind PR, and search for this string "transaction Hash is null", (this file: src/Nethermind/Nethermind.TxPool/Filters/NullHashTxFilter.cs), it is clear that the way they return all fatal errors, is to use the prefix, and add (, reason text xyz).
So seems redundant and confusing to add the specific message in our matching string.

Note that even with this internal string, we still are keeping that open ended match. So adding this detailed string doesn't give us any added protection against side effects.

Reply:
It is not redundant. It is a deliberate design choice to match a limited set instead of an open ended one. I can't say whether it is the correct/best/optimal choice, but it is significant and shouldn't be dismissed hastily.

> Note that even with this internal string, we still are keeping that open ended match. So adding this detailed string doesn't give us any added protection against side effects.

I don't know what you mean. The internal case of `Invalid(, transaction Hash is null)?` is actually not as interesting as I first thought - just another instance of being explicit about the match. We only accept the two literals `Invalid` or `Invalid, transaction Hash is null`.

Reply:
Having `$` at the end of each regex forces an exact match of either one of these two choices. I asked a similar question on slack and it seems like the preferred way is to be as specific as possible.

Reply:
It's a conscious choice, for the reasons I listed here:
https://chainlink-core.slack.com/archives/C0117GGJB6Y/p1669215075401729?thread_ts=1669126977.871589&cid=C0117GGJB6Y